cmd/serverless/api_key.go

```diff
@@ -161,3 +162,20 @@ func hasApiKey() bool {
 		len(os.Getenv(kmsAPIKeyEnvVar)) > 0 ||
 		len(os.Getenv(secretsManagerAPIKeyEnvVar)) > 0
 }
+
+func sendAPIKeyToShell(apiKey string) error {
+	conn, err := net.Dial("tcp", buildShellHostPort())
+	if err != nil {
+		return fmt.Errorf("could not establish connexion to the shell")
+	}
+	n, err := conn.Write([]byte(apiKey))
+	if err != nil || n != len(apiKey) {
+		return fmt.Errorf("error while writing to the shell")
+	}
```

Comment:
You could return the errors directly or just wrap them (eg. `fmt.Errorf("could not establish connexion to the shell: %w", err)`) to avoid losing the real error message.

Reply:
yeah, I think I'll leave it as is as we don't really care about the exact error here